radix-engine/src/system/system_modules/costing/costing_module.rs

```diff
@@ -380,10 +380,8 @@ impl PrivilegedSystemModule for CostingModule {
         api: &mut impl SystemBasedKernelApi,
         invocation: &KernelInvocation<Actor>,
     ) -> Result<(), RuntimeError> {
-        let depth = api.kernel_get_current_depth();
-
         // Skip invocation costing for transaction processor
-        if depth == 0 {
+        if api.kernel_get_system_state().current_call_frame.is_root() {
```

Comment:
MAJOR: I don't think this is right now that we've changed the definition of "root" - I think we need to put this into versioned logic.

Previous, we didn't bill for the Root > Processor call. Now we don't bill for the Processor > XX calls - this seems wrong?

I think the above comment also needs changing.

Reply:
In fact, I imagine this is why this PR changed all the costings. I imagine we can revert them all if we fix this.